src/App.jsx

```diff
@@ -14,45 +17,97 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button
-        type="button"
-        className="button is-info is-light"
-      >
-        Sort alphabetically
-      </button>
-
-      <button
-        type="button"
-        className="button is-success is-light"
-      >
-        Sort by length
-      </button>
-
-      <button
-        type="button"
-        className="button is-warning is-light"
-      >
-        Reverse
-      </button>
-
-      <button
-        type="button"
-        className="button is-danger is-light"
-      >
-        Reset
-      </button>
-    </div>
+export const App = () => {
+  const visibleGoods = [...goodsFromServer];
+
+  const [sortBy, setSortBy] = useState('');
+  const [reverse, setReverse] = useState(false);
+
+  const SORT_BY_ALPH = 'alph';
+  const SORT_BY_LENGTH = 'length';
+
+  switch (sortBy) {
+    case SORT_BY_ALPH:
+      if (reverse) {
+        visibleGoods
+          .sort((good1, good2) => good1.localeCompare(good2)).reverse();
+      } else {
+        visibleGoods
+          .sort((good1, good2) => good1.localeCompare(good2));
+      }
+
+      break;
+
+    case SORT_BY_LENGTH:
+      if (reverse) {
+        visibleGoods
+          .sort((good1, good2) => good1.length - good2.length).reverse();
+      } else {
+        visibleGoods
+          .sort((good1, good2) => good1.length - good2.length);
+      }
+
+      break;
+    default:
+      if (reverse) {
+        visibleGoods.reverse();
+      }
 
-    <ul>
-      <li data-cy="Good">Dumplings</li>
-      <li data-cy="Good">Carrot</li>
-      <li data-cy="Good">Eggs</li>
-      <li data-cy="Good">Ice cream</li>
-      <li data-cy="Good">Apple</li>
-      <li data-cy="Good">...</li>
-    </ul>
-  </div>
-);
+      break;
+  }
+
+  const reset = () => {
+    setSortBy('');
+    setReverse(false);
+  };
+
+  return (
+    <div className="section content">
+      <div className="buttons">
+        <button
+          type="button"
+          className={cn('button is-info',
+            { 'is-light': sortBy !== SORT_BY_ALPH })}
+          onClick={() => setSortBy(SORT_BY_ALPH)}
+        >
+          Sort alphabetically
+        </button>
+
+        <button
+          type="button"
+          className={cn('button is-info',
+            { 'is-light': sortBy !== SORT_BY_LENGTH })}
```

Comment:
You can create a helper variable for this `sortBy !== SORT_BY_LENGTH` and use them everywhere
```suggestion
            { 'is-light': sortBy !== SORT_BY_LENGTH })}
```